bundles/pcmumlclass/tools.vitruv.applications.pcmumlclass/src/tools/vitruv/applications/pcmumlclass/PcmUmlClassHelper.xtend

```diff
@@ -22,14 +22,61 @@ import java.util.function.Function
 import org.eclipse.emf.ecore.resource.Resource
 import tools.vitruv.framework.userinteraction.UserInteractor
 import tools.vitruv.framework.userinteraction.UserInteractionOptions.NotificationType
+import org.eclipse.uml2.uml.Model
+import java.util.Set
+import org.apache.log4j.Logger
+import tools.vitruv.extensions.dslsruntime.reactions.helper.PersistenceHelper
+import edu.kit.ipd.sdq.commons.util.org.eclipse.emf.common.util.URIUtil
+import java.util.Optional
 
 class PcmUmlClassHelper {
 	private static val PCM_PRIMITIVE_TYPES_URI = URI.createURI("pathmap://PCM_MODELS/PrimitiveTypes.repository");
 	private static val UML_PRIMITIVE_TYPES_URI = URI.createURI("pathmap://UML_LIBRARIES/UMLPrimitiveTypes.library.uml");
+	 private static val logger = Logger.getLogger(PcmUmlClassHelper.simpleName)
 
 	private new() {
 	}
 
+	def static Optional<Model> loadPersistedModelFromSource(String relativeModelPath, EObject source) {
+		var validatedModelPath = relativeModelPath
+		if (validatedModelPath.nullOrEmpty) {
+			validatedModelPath = DefaultLiterals.MODEL_DIRECTORY + "/" + DefaultLiterals.UML_MODEL_FILE_NAME + DefaultLiterals.UML_EXTENSION;
+		} else if (!validatedModelPath.endsWith(DefaultLiterals.UML_EXTENSION)) {
+			validatedModelPath += DefaultLiterals.UML_EXTENSION
+		}
+		//check if a model at the specified path already exists:
+		val uri = PersistenceHelper.getURIFromSourceProjectFolder(source, relativeModelPath) 
+		if (URIUtil.existsResourceAtUri(uri)) { 
+			// The resource is only found if it was previously persisted, which only happens after the change propagation terminates.
+			// This should not be a problem, as long as any created model is registered on creation, to prevent creating a second one.
+			val resource = source.eResource.resourceSet.getResource(uri, true)
+			return Optional.of(resource.contents.filter(Model).head)
+		}
+		return Optional.empty
+	}
+
+	/**
+	 * TODO FIXME TS This method is DIRECTLY copied from  tools.vitruv.applications.umljava.java2uml.JavaToUmlHelper due to the lack of a shared helper/util class
```

Comment:
This is part of a larger issue we might need to talk about. Right now, there is no way of sharing helper methods across reactions that have the same target model without using unreasonable dependencies (in this case PCM to UML and Java to UML). I think we briefly talked about that before.